Added additional WellKnownSidType enumeration members.

couven92 opened the pull request:
Documentation copied and adapted from https://msdn.microsoft.com/en-us/library/aa379650(v=vs.140).aspx

karelz commented:
Be warned that /// comments are not flowing into Intellisense or documentation yet. We plan to finalize the plan around it in Feb/March.

@steveharter @bartonjs any advice on who should code review? (I guess this is part of the grey area)

bartonjs commented on src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs:
```diff
@@ -50,68 +50,245 @@ internal enum SidNameUse
         BuiltinAuthorizationAccessSid = 59,
+        /// <summary>Indicates a SID is present in a server that can issue terminal server licenses.</summary>
         WinBuiltinTerminalServerLicenseServersSid = 60,
-        MaxDefined = WinBuiltinTerminalServerLicenseServersSid,
```
@terrajobst Looks like the thing which uses MaxDefined in a validator is the product code.  https://referencesource.microsoft.com/#mscorlib/system/security/principal/sid.cs,491, and without updating it (in the impl) it'll make the new values unusable.

A) Take the breaking change of updating it
B) Take the breaking change of undefining it
C) Update it in the impl, but not the ref
D) Say we're never allowed to update this enumeration
E) Leave this value as-is, but change the product code to do something smarter.
F) Some other proposal.

cc @weshaggard, @ericstj

danmosemsft replied:
@weshaggard, @ericstj ?

terrajobst commented:
We probably shouldn't update the value of `MaxDefined`. It's unfortunate that it shipped because our guidelines explicitly call out not to use sentinel values in enums. And changing enum values is technically a binary breaking change.

bartonjs commented on src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs:
```diff
@@ -50,68 +50,245 @@ internal enum SidNameUse
+        WinMediumPlusLabelSid = 79,
+        /// <summary>
+        /// Indicates a SID that matches a local logon group.
+        /// <para><strong>Windows Server�2008�R2, Windows�7, Windows Server�2008, Windows�Vista, Windows Server�2003 and Windows�XP:</strong> This value is not available.</para>
```
This comment doesn't correctly apply.  For native it means "you'll get a compile failure if you compile for Windows 7".  For .NET there'd be no such compile failure.

If Windows 7 with .NET Core called `SecurityIdentifier..ctor(WellKnownSidType, SecurityIdentifier)` they'd get an ArgumentException indicating either sidType or domainSid was invalid (quite specific, those being the only two parameters to the ctor), which is fairly close to the current/.NET Framework behavior of ArgumentException that it's not a defined sidType value.

So I think the right answer is to just drop all the comments to the one-sentence summary.

couven92 replied:
Totally agree... I simply blindly copied the documentation from the MSDN pages.

Changed in commit f59853c003ff22c557f9d8f5ec5817997b2eaaf2

couven92 commented:
@terrajobst 
> We probably shouldn't update the value of MaxDefined.
Does that mean that you want me to leve the `MaxDefined` member as it originnaly was, and simply add the new values after it?

karelz commented:
It means we need to review the change from API/compat perspective - I created #14636 to track it (PR is bad place to have design discussion).
We need to block the PR on the decision - just a heads up that it may take a while, we take compat breaking changes seriously ...

karelz commented:
Further guidance on what to do is in https://github.com/dotnet/corefx/issues/14636#issuecomment-268330861

couven92 commented:
Sorry about the duplicate commit messages, I messed up the push and had to force push to correct. So commit c838dc7 is the only that actually ended up on the branch.

karelz commented on src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs:
```diff
@@ -50,68 +50,197 @@ internal enum SidNameUse
+        /// <summary>Indicates a SID is present in a server that can issue terminal server licenses.</summary>
         WinBuiltinTerminalServerLicenseServersSid = 60,
-        MaxDefined = WinBuiltinTerminalServerLicenseServersSid,
+        MaxDefined = WinBuiltinTerminalServerLicenseServersSid, 
```
We need to add `[Obsolete]` + `[EditorBrowsable(EditorBrowsableState.Never)]`.

couven92 replied:
Done in commit a35dcb2

karelz commented on src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs:
```diff
@@ -50,68 +51,198 @@ internal enum SidNameUse
+        /// <summary>Indicates a SID is present in a server that can issue terminal server licenses.</summary>
         WinBuiltinTerminalServerLicenseServersSid = 60,
-        MaxDefined = WinBuiltinTerminalServerLicenseServersSid,
+        [Obsolete, EditorBrowsable(EditorBrowsableState.Never)]
```
Nit: Could you please split it into 2 lines? It's easier to read + AFAIK it is the general unwritten rule in the code base ...
```c#
[Obsolete]
[EditorBrowsable(EditorBrowsableState.Never)]
```

couven92 replied:
Commit 4d30256 

Merry Christmas everyone!

danmosemsft commented:
@bartonjs @karelz can you please re-review? @couven92 says this is good to go.

bartonjs commented:
This still isn't ready to go.  Aside from the API review happening, all of the points I made in https://github.com/dotnet/corefx/pull/14611#pullrequestreview-13607431 are still outstanding.

* Tests using the new values are required.
* This edits the implementation definition of the enumeration, but not the contract/ref version, so no one can use it.
* No acknowledgement was made on whether any of the new values require extra validation (like some of the existing values do).

Also, there appear to be some alignment errors with the attributes on `MaxDefined` (they aren't indented the same).

couven92 commented:
Okay, with commit 4f62573 I have added a Unit Test that attempts to construct a `SecurityIdentifier` using values from the `WellKnownSidType` enumeration. I made it a theory as there are values (e.g. `WellKnownSidType.LogonIdsSid`) where the SecurityIdentifier constructor legitimately throws an argument exception.

With commit 491a6ac  I have also updated the ref source.

Commit 75477e6 fixes the inconsistent whitespace, sorry about that... I did that in a rush shortly before my plane was landing and didn't check everything properly...

karelz commented:
@couven92 please let us know when you think you are done and your PR is ready for another round of review.

karelz commented on src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs:
```diff
@@ -50,68 +51,199 @@ internal enum SidNameUse
-        MaxDefined = WinBuiltinTerminalServerLicenseServersSid,
+        [Obsolete]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        MaxDefined = WinBuiltinTerminalServerLicenseServersSid, 
```
Can you please add a comment to explaining why it is obsolete: That it was created originally by mistake and CoreFX should not rely on it anymore, but we have to keep it for compatibility.

couven92 replied:
Commit 4af4650

couven92 replied:
Please have a look at the Obsoletion message. I think it is explanatory enough, but maybe you think it's not verbose enough or maybe too verbose?
`"This member has been depcreated and is only maintained for backwards compatability. WellKnownSidType values greater than MaxDefined may be defined in future releases."`

karelz commented on src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs:
```diff
@@ -50,68 +51,199 @@ internal enum SidNameUse
+        /// <summary>Indicates a SID for Windows credentials capability for app containers.</summary>
+        WinCapabilityEnterpriseAuthenticationSid = 93,
+        /// <summary>Indicates a SID for removable storage capability for app containers.</summary>
+        WinCapabilityRemovableStorageSid = 94
```
Can you please add comment to the end, saying that the code below has to be updated whenever we add a new value?

couven92 replied:
Commit 4af4650

karelz commented on src/System.Security.Principal.Windows/tests/System.Security.Principal.Windows.Tests.csproj:
```diff
@@ -16,6 +16,7 @@
   <ItemGroup>
     <Compile Include="WindowsIdentityTests.cs" />
     <Compile Include="WindowsPrincipalTests.cs" />
+	<Compile Include="WellKnownSidTypeTests.cs" />
```
Nit: Weird spacing

couven92 replied:
Commit a4315a3

karelz commented on src/System.Security.Principal.Windows/tests/WellKnownSidTypeTests.cs:
```diff
@@ -0,0 +1,114 @@
+
+        Assert.True(wellKnownSidInstance.IsWellKnown(sidType));
+    }
+}
```
Can you please add negative test case to test WinCapabilityRemovableStorageSid+1 (and probably also -1)?

Also I would add compat test MaxDefined == WinBuiltinTerminalServerLicenseServersSid. To prevent accidental changes to it in future.

karelz commented:
It looks like the ref changes are not truly consumed by test build:
https://ci.dot.net/job/dotnet_corefx/job/master/job/windows_nt_debug_prtest/1912/
`error CS0117: 'WellKnownSidType' does not contain a definition for 'WinBuiltinDCOMUsersSid' [D:\j\workspace\windows_nt_de---4526f5ff\src\System.Security.Principal.Windows\tests\System.Security.Principal.Windows.Tests.csproj]`

@ericstj @weshaggard @mellinoe any idea what's wrong here?

weshaggard commented:
>@ericstj @weshaggard @mellinoe any idea what's wrong here?

In order for the test projects to see the live builds of the ref they need to temporarily take a project reference to the pkgproj.

karelz commented:
Wow, that is complicated (it feels like I learn a new thing in the process every week :(). Do we have it documented anywhere?
Will it be better in the new infra?

couven92 commented:
@weshaggard I did not really understand that... Is there something I have to do? And if yes, how do I do it?

weshaggard commented on src/System.Security.Principal.Windows/ref/System.Security.Principal.Windows.cs:
```diff
@@ -114,68 +114,201 @@ public enum TokenAccessLevels
+        [System.ObsoleteAttribute("This member has been depcreated and is only maintained for backwards compatability. WellKnownSidType values greater than MaxDefined may be defined in future releases.")]
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        MaxDefined = WinBuiltinTerminalServerLicenseServersSid,
+        #if NETSTANDARD20
```
Why did you add this #ifdef? That is what is causing the compilation error. I think you should just remove it all together.

couven92 replied:
Ah misunderstanding.... https://github.com/dotnet/corefx/issues/14636#issuecomment-270160041 suggested the `#ifdef`. But let's remove it and see what happens...

weshaggard commented:
Actually looking at the test project it already has that reference.

```
     <ProjectReference Include="..\pkg\System.Security.Principal.Windows.pkgproj">
       <Project>{f9e9894e-2513-4085-9046-311ad49d8ae6}</Project>
       <Name>System.Security.Principal.Windows</Name>
     </ProjectReference>
```

which is what I was referring to, so that isn't the issue. 

The compile issue is caused by the #ifdef I added a comment on.

weshaggard commented:
```
D:\j\workspace\windows_nt_de---4526f5ff\Tools\ApiCompat.targets(68,5): error : MembersMustExist : Member 'System.Security.Principal.WellKnownSidType System.Security.Principal.WellKnownSidType.WinAccountReadonlyControllersSid' does not exist in the implementation but it does exist in the contract. [D:\j\workspace\windows_nt_de---4526f5ff\src\System.Security.Principal.Windows\src\System.Security.Principal.Windows.csproj]
```

The ApiCompat errors are because you have added new members but they don't exist in net463. So to fix for now we should just baseline those compat issues.  To do that create a baseline file similar to https://github.com/dotnet/corefx/blob/master/src/System.AppContext/src/ApiCompatBaseline.net463.txt in your src directory and add all the issues APICompat is complaining about. I generally copy them from the log and clean-up the spacing at beginning and end of each line.

couven92 commented:
@weshaggard Like this? 7af7978dd986d5ba11457f74388e59e659641006

weshaggard commented:
Yes that looks like what I was expecting. Fingers crossed that CI will like it :)

bartonjs commented:
@weshaggard Shouldn't the new enum members be in a netcoreapp contract only (not netstandard17)?

weshaggard commented:
I missed this earlier but you put the baseline file at `src/System.Security.Principal.Windows/ApiCompatBaseline.net463.txt` but it should be at `src/System.Security.Principal.Windows/src/ApiCompatBaseline.net463.txt`.

>@weshaggard Shouldn't the new enum members be in a netcoreapp contract only (not netstandard17)?

Yes your right they probably should be (I'm in a different mindset as I've been working in dev/eng too much recently). To make these netcoreapp specific you need to create a new build configuration for both the ref and src. @couven92 I will pull down your changes provide a change you can pick-up which should fix this.

couven92 commented:
@weshaggard sure, be my guest! I'll move the Basefile in the meantime

weshaggard commented:
@couven92 I've push a commit to branch https://github.com/weshaggard/corefx/commit/7d70b17a2cac5f51ca2541016c670ab11c271e52 which changes your projects to be netcoreapp1.1 specific instead of netstandard1.7 which is what I think we should have for this. They all build now but there are still some test failures that I will let you investigate from this point.  You can cherry-pick my commit and just do a `msbuild /t:rebuildandtest` in the test directory to see the failures.

couven92 commented:
@weshaggard Well, now with commit 40ef6a8, all tests that run from inside `src\System.Security.Principal.Windows\tests` pass.

However, if I run `build.cmd` I get one error message:
`.\Tools\Packaging.targets(1099,5): error : System.Security.Principal.Windows should support API version 4.1.0.0 on .NETStandard,Version=v1.7 but .\packages\System.Security.Principal.Windows\4.3.0\ref\netstandard1.3\System.Security.Principal.Windows.dll was found to support 4.0.0.0. [.\src\System.Security.Principal.Windows\pkg\System.Security.Principal.Windows.pkgproj]`

weshaggard commented:
Sorry I missed that, cherry-pick commit https://github.com/weshaggard/corefx/commit/814ccf20e5dde60a98608c26b364862d070cba83 to address that validation error which isn't even valid so we will just suppress it.

karelz commented:
CI seems to have some infra troubles on all platforms:

Failure: https://ci.dot.net/job/dotnet_corefx/job/master/job/centos7.1_debug_prtest/1955/
> There was an error processing the xunit test results: [xUnit] [ERROR] - No test reports found for the metric 'xUnit.Net' with the resolved pattern 'bin/tests/**/testResults.xml'. Configuration error?.
(same on all platforms)

@mmitche @MattGal is it known issue?

karelz commented:
@dotnet-bot test this please

MattGal commented:
It’s fairly clear from the logs that tests did not start so the Jenkins thing trying to match the result XMLs fails.   Whichever change brought these errors likely caused it:

10:03:56 /mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/Tools/Packaging.targets(1099,5): error : System.Security.Principal.Windows should support API version 4.1.0.0 on .NETStandard,Version=v1.7 but /mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/packages/System.Security.Principal.Windows/4.3.0/ref/netstandard1.3/System.Security.Principal.Windows.dll was found to support 4.0.0.0. [/mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/src/System.Security.Principal.Windows/pkg/System.Security.Principal.Windows.pkgproj]
10:03:56 /mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/Tools/Packaging.targets(1099,5): error : System.Security.Principal.Windows should support API version 4.1.0.0 on .NETStandard,Version=v2.0 but /mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/packages/System.Security.Principal.Windows/4.3.0/ref/netstandard1.3/System.Security.Principal.Windows.dll was found to support 4.0.0.0. [/mnt/resource/j/workspace/dotnet_corefx/master/centos7.1_debug_prtest/src/System.Security.Principal.Windows/pkg/System.Security.Principal.Windows.pkgproj]

I need to pack my office for paint at the moment and be out of here by 5 PM, but trolling history of that package should find the culprit.

-Matt

From: Karel Zikmund [mailto:notifications@github.com]
Sent: Monday, January 9, 2017 4:17 PM
To: dotnet/corefx <corefx@noreply.github.com>
Cc: Matt Galbraith <Matt.Galbraith@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [dotnet/corefx] Added additional WellKnownSidType enumeration members. (#14611)


CI seems to have some infra troubles on all platforms:

Failure: https://ci.dot.net/job/dotnet_corefx/job/master/job/centos7.1_debug_prtest/1955/<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fci.dot.net%2Fjob%2Fdotnet_corefx%2Fjob%2Fmaster%2Fjob%2Fcentos7.1_debug_prtest%2F1955%2F&data=02%7C01%7CMatt.Galbraith%40microsoft.com%7C9d85e2b6984743ab31d508d438edf711%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636196042153086256&sdata=nYuI8hUrn%2FSnLBysAd1mxD%2FVDqnHPBvV7o81XTFEOfU%3D&reserved=0>

There was an error processing the xunit test results: [xUnit] [ERROR] - No test reports found for the metric 'xUnit.Net' with the resolved pattern 'bin/tests/**/testResults.xml'. Configuration error?.
(same on all platforms)

@mmitche<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmmitche&data=02%7C01%7CMatt.Galbraith%40microsoft.com%7C9d85e2b6984743ab31d508d438edf711%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636196042153086256&sdata=i7i2WaO2wjP5flas%2BgC6zxXt6xhWmOaXlNpBF8jPIeY%3D&reserved=0> @MattGal<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMattGal&data=02%7C01%7CMatt.Galbraith%40microsoft.com%7C9d85e2b6984743ab31d508d438edf711%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636196042153086256&sdata=udw9igwvB1Z3%2FHl%2BfGulVA2GRp%2BG%2BPZZoHjHrLXd2JQ%3D&reserved=0> is it known issue?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fcorefx%2Fpull%2F14611%23issuecomment-271449299&data=02%7C01%7CMatt.Galbraith%40microsoft.com%7C9d85e2b6984743ab31d508d438edf711%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636196042153086256&sdata=L8Up8crRkyKhDuxbzw8NY%2Bdfh5Cb%2B%2FG01zmThPOvMEU%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAHf2-kDTwiqVa_d79KhZFxg0FZ5IRMEgks5rQs3sgaJpZM4LQCgV&data=02%7C01%7CMatt.Galbraith%40microsoft.com%7C9d85e2b6984743ab31d508d438edf711%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636196042153086256&sdata=FwoxL7BP4Smh5y1EqoyUmqHMHKf%2BZvgwvcv5l2ndq0c%3D&reserved=0>.

karelz commented:
Ah, I didn't realize there were errors above the one linked. Thanks @MattGal! On a side note: Could we improve Jenkins to show all errors from the log? That would make things much easier.

This definitely looks related to the changes in this PR. @weshaggard any further guidance?

weshaggard commented:
The error that @couven92 called out should be fixed by the commit I pointed at (https://github.com/weshaggard/corefx/commit/814ccf20e5dde60a98608c26b364862d070cba83). So once @couven92 has time to cherry-pick that commit that error should go away.

karelz commented:
Ah, sorry - I missed that it was not applied yet.

couven92 commented:
Sorry for the delay. Sth. came up, so I became unavailable.

But now I have cherry-picked @weshaggard commit.

karelz commented:
@dotnet-bot test this please

weshaggard commented:
I'm doing a full build locally to try and help with the current build issue...

weshaggard commented:
OK I've made another commit that I believe should fix the remaining issues. https://github.com/weshaggard/corefx/commit/dd2694721f0c4fb924ec5d7efbce1b27773b6208.

On my local machine some of your tests are still failing for me because current https://github.com/dotnet/corefx/pull/14611/files#diff-ba321b3c379beee0218e2ecf0a28aec2R107, which causes some of the SID checks to fail. I believe for some reason that is null on my local runs because I'm running them as admin. If I run them as non-admin they pass. I don't know enough about the identity space to understand why that is the case but we should try and make sure all our tests pass correctly in both admin and non-admin mode, even if they are partitioned in some way between the two.

weshaggard commented:
By the way @couven92 I would like to thank you for your contributions and patience as we work through this, we definitely know it is way harder then it should be right now. We are doing a fair amount of work in our dev/eng branch which will make this type of change much easier in the future. That work should hopefully make it into master in the coming weeks.

karelz commented:
Ditto on thanks for patience from @couven92! And huge thanks to @weshaggard for help here and for making it better in dev/eng - I can't wait to see the better stuff to be in :).

karelz commented:
@couven92 do you plan to finish the PR? (no rush, just checking)

couven92 commented:
@karelz Sorry, with the dev-eng changes, I chose to wait for the storm to blow over first, but I got a little tied up with the start of the new semester at my university in this week. I'll update the source to be mergeable with the master branch now, and then I'll run the tests again to see what is remaining...

@weshaggard Yeah, thanks for the support! This is my first PR in a repo of such a large project like .NET. It has been really educational. And the feedback I get from you guys is great, making this really fun!! Definitely going to do more PRs in the future when the occasion arises!!! :)

couven92 commented on src/System.Security.Principal.Windows/tests/System.Security.Principal.Windows.Tests.builds:
```diff
@@ -4,7 +4,7 @@
     <Project Include="System.Security.Principal.Windows.Tests.csproj">
       <OSGroup>Windows_NT</OSGroup>
-      <TestTFMs>netcore50;netcoreapp1.0;net46;netcoreapp;net463</TestTFMs>
+      <TestTFMs>netcore50;netcoreapp1.0;net46;netcoreapp1.1;net463</TestTFMs>
```
@weshaggard I changed this when I merged with the master (after the dev-eng work)... `netcoreapp1.1` seemed like the more logical choice, am I right?

weshaggard replied:
This file should actually be deleted now as we have deleted all ".builds" as we aren't using them after the engineering change.

karelz commented:
@couven92 thanks for confirmation - as I said, no rush.
Waiting for the storm to blow over is usually smart move. The good news is that this particular storm didn't cause much damage, so hopefully it won't cause you too much additional work.

couven92 commented:
@karelz @weshaggard Oho: All checks have passed

This looks promising...

couven92 commented:
I see there is another build file in `src\System.Security.Principal.Windows\pkg\System.Security.Principal.Windows.builds` Delete that one too?

karelz commented:
It looks independent on this PR, which is already passing CI after all. Let's deal with the deletion of the extra .builds file in separate PR, if it is needed.
Merging ...

karelz commented:
Actually, while creating the squash & merge commit message, I have noticed you deleted 2 test cases - what is the reason for not having them?
We should probably leave them commented out with explanation why they are commented out - imagine that someone will try to review the test case in a few months/years and notices 2 cases missing - we should prevent wasting that person's time on rediscovering why they are missing / avoid attempts to add them back.

couven92 commented:
Ah, yes, I am not entirely sure, but I think both `WinLocalLogonSid` and `WinApplicationPackageAuthoritySid` are some kind of special case authority SID types. I got ArgumentExceptions no matter what domain SID I passed to the SecurityIdentifier constructor.

From my understanding `WinLocalLogonSid` and `WinApplicationPackageAuthoritySid` should be constructable from the `NullSid` domain SID, but that didn't work either.

I have added comments there as you suggested @karelz, and I will explore the two corner cases later and create a new PR for missing test cases then.

couven92 commented:
No, I have played around with `WinLocalLogonSid` and `WinApplicationPackageAuthoritySid` trying to create a `SecurityIdentifier` instance.

Neither of these work:
- Setting `domainSid` to `null`
- Setting `domainSid` to one of the following constructed Well-Known SIDs:
  1. `NullSid`
  2. `WorldSid`
  3. `LocalSid`
  4. `NTAuthoritySid`
In all cases I get the `ArgumentException` that is thrown when the Win32 P/Invoke call to Advapi32.dll returns an `ERROR_INVALID_PARAMETER`

What's interesting about that is that `WinLocalLogonSid` throws an exception, but `WinConsoleLogonSid` does not.

Anyone any ideas what is up with `WinLocalLogonSid` and `WinApplicationPackageAuthoritySid`?

couven92 commented:
Hmm... and when I use the binary form of the App Package Authority SID (as documented on https://msdn.microsoft.com/en-us/library/hh448449.aspx the SecurityIdentifier is built, but then the Assertion that it is the WellKnownSidType `WinApplicationPackageAuthoritySid` fails...

karelz commented:
Thanks for adding the 2 commented out test cases. If they are problematic, we can totally handle them in separate PR, or drop them :). The comment will be useful for the next person looking at the code / test.
I'll try to squash & merge later today.

karelz commented:
Thanks @couven92 for your contribution and for your patience!

weshaggard commented:
Thanks @couven92 for your patience and contribution.

I did want to follow-up on some of my earlier questions about running the tests in admin mode.

>On my local machine some of your tests are still failing for me because current https://github.com/dotnet/corefx/pull/14611/files#diff-ba321b3c379beee0218e2ecf0a28aec2R107, which causes some of the SID checks to fail. I believe for some reason that is null on my local runs because I'm running them as admin. If I run them as non-admin they pass. I don't know enough about the identity space to understand why that is the case but we should try and make sure all our tests pass correctly in both admin and non-admin mode, even if they are partitioned in some way between the two.

Were you able see what that issue is? I suspect more folks will start hitting this at some point so it will be good to get ahead of it.

karelz commented:
@weshaggard can you please file a bug? Ideally with the list of failed tests.

weshaggard commented:
I filed issue https://github.com/dotnet/corefx/issues/15436.
